### DISCUSSION TITLE
Shifting from C to TypeScript: Developer Benefits

### DISCUSSION THREAD

**Discussion by pacifiquem** [kin-lang/kin#30] [General]
The reason why we migrated from C to TypeScript in #29.

Why C?
======

Why we choose C in the first place.

- Efficient Memory Management: C allowed us to manage memory by ourselves this allowed us to have Precision in memory control.
- Compilation and Stability : C being a compiled programming language allowed us to build and distrubute interpreter's binaries easly
- Other points:
	- Low Level
	- Availability of resources

Why TypeScript Now?
=================

Why are we going to use TypeScript now?

- Growing Community Engagement : This is the main point. Growing Community! As we all know, open source is pushed by collaborators and having Kin written in C didn't favor us since many collaborators we expect are those from Rwanda and C is not known by the majority of the developers.
- Compatibility with Web Online Editors : Since the only editor we'll have for Kin will be web-based, TypeScript fits perfectly. There will be no build problems with it hence it's in our favor
- Other reasons:
	- Object-Oriented Programming (OOP) Support

This shift aims to embrace a wider community, streamline development with OOP support, and enhance collaboration with web-based tools.

**Comment by Hallcoder** [comment]
Yeah so true, let's hope the TS developers join the movement atleast , you can rewrite it in Rust later 😅😅
